modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/ProcessStore.java

```diff
@@ -53,11 +54,129 @@
 import java.util.zip.ZipInputStream;
 
 public class ProcessStore {
-
-    private static final Log log = LogFactory.getLog(ProcessStore.class);
-
-    private static final String mns = "http://www.wso2.org/governance/metadata";
-    private static final String OK = "OK";
+	private static final Log log = LogFactory.getLog(ProcessStore.class);
+
+	private static final String mns = "http://www.wso2.org/governance/metadata";
+	private static final String OK = "OK";
+
+	public String getProcessVariablesList(String resourcePath) {
+		String resourceString = "";
+		try {
+			RegistryService registryService =
+					ProcessCenterServerHolder.getInstance().getRegistryService();
+			if (registryService != null) {
+				UserRegistry reg = registryService.getGovernanceSystemRegistry();
+				resourcePath = resourcePath.substring(ProcessStoreConstants.GREG_PATH.length());
+				Resource resourceAsset = reg.get(resourcePath);
+				String resourceContent = new String((byte[]) resourceAsset.getContent());
+
+				JSONObject conObj = new JSONObject();
+				DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+				DocumentBuilder builder;
+				builder = factory.newDocumentBuilder();
+				Document document =
+						builder.parse(new InputSource(new StringReader(resourceContent)));
+
+				JSONArray variableArray = new JSONArray();
+
+				conObj.put("processVariables", variableArray);
+
+				NodeList processVariableElements =
+						((Element) document.getFirstChild()).getElementsByTagName("process_variable");
+
+				if (processVariableElements.getLength() != 0) {
+					for (int i = 0; i < processVariableElements.getLength(); i++) {
+						Element processVariableElement = (Element) processVariableElements.item(i);
+						String processVariableName =
+								processVariableElement.getElementsByTagName("name").item(0)
+										.getTextContent();
+						String processVariableType =
+								processVariableElement.getElementsByTagName("type").item(0)
+										.getTextContent();
+
+						JSONObject processVariable = new JSONObject();
+						processVariable.put("name", processVariableName);
+						processVariable.put("type", processVariableType);
+						variableArray.put(processVariable);
+					}
+				}
+				resourceString = conObj.toString();
+			}
+		} catch (Exception e) {
```

Comment:
Thanks @cnapagoda . Initially there was a 'Exception' throwing function call in the try block and that is why it had been caught. But later that function call had been removed, so yes it should not been caught now. So fixed with 0191c09.